pep-perennial.rst

```diff
@@ -0,0 +1,178 @@
+use of these features. They define a set of core libraries and symbol versions
+which wheels may expect on the system, based on CentOS 5 and 6 respectively.
+Typically, packages are built in Docker containers based on these old CentOS
+versions, and then the ``auditwheel`` tool is used to check them and bundle any
+other linked libraries into the wheel.
+
+If we were to define a ``manylinux2014`` tag based on CentOS 7, there would be
+five steps involved to make it practically useful:
+
+1. Write a PEP
+2. Prepare docker images based on CentOS 7.
+3. Add the definition to auditwheel
+4. Allow uploads with the new tag on PyPI
+5. Add code to pip to recognise the new tag and check if the platform is
+   compatible
+
+Although preparing the docker images and updating auditwheel take more work,
+these parts can be used as soon as that work is complete. The changes to pip
+are more straightforward, but not all users will promptly install a new version
+of pip, so distributors are concerned about moving to a new tag too quickly.
+
+This PEP aims to remove the need for steps 1 and 5 above, so new manylinux tags
+can be adopted more easily.
+
+Naming
+======
+
+Tags using the new scheme will look like::
+
+    manylinux_glibc_2_17_x86_64
+
+Where ``2_17`` is the major and minor version of glibc. I.e. for this example,
+the platform must have glibc 2.17 or newer. Installer tools should be prepared
+to handle any numeric values here, but building and publishing wheels to PyPI
+will probably be constrained to specific profiles defined by auditwheel.
+
+The existing manylinux tags can also be represented in the new scheme:
+
+- ``manylinux1_x86_64`` becomes ``manylinux_glibc_2_5_x86_64``
+- ``manylinux2010_x86_64`` becomes ``manylinux_glibc_2_12_x86_64``
+
+``x86_64`` refers to the CPU architecture, as in previous tags.
+
+While this PEP does not attempt to define tags for non-glibc Linux, the name
+glibc is included to leave room for future efforts in that direction.
+
+Wheel compatibility
+===================
+
+There are two components to a tag definition: a specification of what makes a
+compatible wheel, and of what makes a compatible platform.
+
+A wheel may never use symbols from a newer version of glibc than that indicated
+by its tag. Likewise, a wheel with a glibc tag under this scheme may not be
+linked against another libc implementation.
+
+As with the previous manylinux tags, wheels will be allowed to link against
+a limited set of external libraries and symbols. These will be defined by
+profiles in auditwheel. At least initially, they will likely be similar to
+the list for manylinux2010 (:pep:`571`), and based on library versions in
+newer versions of CentOS.
+
+The overall goal is to ensure that if a wheel is tagged as
+``manylinux_glibc_2_Y``, then users can be reasonably confident that this wheel
+will work in any real-world linux-based python environment that uses
```

Comment:
I am slightly scared of the `any real-world linux-based python environment` expression.
This may make it virtually limitless, and make defining new profiles extremely difficult.
How about creating a list here with a few big/old distros, such as debian, ubuntu, RHEL, centOS, Arch, Suse, and make the profile creation more tractable?

Or we can fork this, and have some official profiles that are "promised" to be automatically generated, but then have community create more auxiliary profiles?

Reply:
And another concern, would having usch profiles make auditwheel checks longer in time?